lib/Dancer2/Manual/Testing.pod

```diff
@@ -49,8 +49,104 @@ A basic test (which we also scaffold with L<dancer2>) looks like this:
 Read the documentation for L<HTTP::Request> and L<HTTP::Request::Common> to
 see the different options for sending parameters.
 
+=head3 Subtests
+
+Tests can be separated using L<Test::More>'s C<subtest> functionality,
+thus creating multiple self-contained tests that don't overwrite each other.
+
+Assuming we have a different app that has two states we want to test:
+
+     # MyApp.pm
+     package MyApp;
+     use Dancer2;
+     set serializer => 'JSON';
+
+     get '/:user' => sub {
+         my $user = route_parameters->get('user');
+
+         $user and return { user => $user };
+
+         return {};
+     };
+
+     1;
+
+This is a contrived example of a route that checks for a user
+parameter. If it exists, it returns it in a hash with the key
+'user'. If not, it returns an empty hash
+
+     # param.t
+     use strict;
+     use warnings;
+     use Test::More;
+     use Plack::Test;
+     use HTTP::Request::Common;
+     use MyApp;
+
+     my $test = Plack::Test->create( MyApp->to_app );
+
+     subtest 'A empty request' => sub {
+         my $res = $test->request( GET '/' );
+         ok( $res->is_success, 'Successful request' );
+         is( $res->content '{}', 'Empty response back' );
+     };
+
+     subtest 'Request with user' => sub {
+         my $res = $test->request( GET '/?user=sawyer_x' );
```

Comment:
oops. The code we're testing is using  route param, not a query param. I sent a PR with the example modified.

Reply:
I pushed a second coming on #1726. That should do it.